### PR TITLE
add support for mutual TLS authentication (client certificates)

### PR DESCRIPTION
**Pull request by hexagonal-sun**
**Description**

This PR implements mutual TLS auth.  Two new options have been created `client_cert` and `client_key`.  These allow the respective PEM files to be passed to `glab` which then allows us to use it for mutual TLS auth.

(Apologies for any obvious errors here, I've never coded in `go` before!)

**Related Issue**
N/A

**How Has This Been Tested?**
I've ran this against my local gitlab instance and it works fine.

**Screenshots (if appropriate):**

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

File: api/client.go [hunk 1]
```diff
@@ -178,6 +178,56 @@ func NewClientWithCustomCA(host, token, caFile string, isGraphQL bool) (*Client,
...
+					KeepAlive: 30 * time.Second,
+				}).DialContext,
+				ForceAttemptHTTP2:     true,
+				MaxIdleConns:          100,
+				IdleConnTimeout:       90 * time.Second,
+				TLSHandshakeTimeout:   10 * time.Second,
+				ExpectContinueTimeout: 1 * time.Second,
+				TLSClientConfig: &tls.Config{
+					RootCAs: caCertPool,
+                    Certificates: clientCerts,
```
Comment (profclems) [inline api/client.go]: This should be `gofmt`'ed. 

Please run `make fix` before pushing

**Review by profclems — CHANGES_REQUESTED** [review]
Thanks for working on this @hexagonal-sun.

This looks good except that it's not `gofmt`'ed.

**Comment by hexagonal-sun** [comment]
@profclems Oops, sorry, I missed the `gofmt` step.  That should now be fixed.

**Review by profclems — APPROVED** [review]
Great! Looks good...

Thanks fir working on this @hexagonal-sun

**Pull request merged**